core/lib/zksync_core/src/db_pruner/mod.rs

```diff
@@ -0,0 +1,502 @@
+            return Ok(false);
+        }
+
+        let next_miniblock_to_prune = transaction
+            .blocks_dal()
+            .get_miniblock_range_of_l1_batch(next_l1_batch_to_prune)
+            .await?
+            .unwrap()
+            .1;
+        transaction
+            .pruning_dal()
+            .soft_prune_batches_range(next_l1_batch_to_prune, next_miniblock_to_prune)
+            .await?;
+
+        transaction.commit().await?;
+
+        let latency = latency.observe();
+        tracing::info!(
+            "Soft pruned db l1_batches up to {} and miniblocks up to {}, operation took {:?}",
+            next_l1_batch_to_prune,
+            next_miniblock_to_prune,
+            latency
+        );
+
+        Ok(true)
+    }
+
+    async fn hard_prune(&self, pool: &ConnectionPool<Core>) -> anyhow::Result<()> {
+        let latency = METRICS.pruning_chunk_duration[&MetricPruneType::Hard].start();
+
+        let mut storage = pool.connection().await?;
+        let mut transaction = storage.start_transaction().await?;
+
+        let current_pruning_info = transaction.pruning_dal().get_pruning_info().await?;
+        transaction
+            .pruning_dal()
+            .hard_prune_batches_range(
+                current_pruning_info.last_soft_pruned_l1_batch.unwrap(),
+                current_pruning_info.last_soft_pruned_miniblock.unwrap(),
+            )
+            .await?;
+
+        transaction.commit().await?;
+
+        let latency = latency.observe();
+        tracing::info!(
+            "Hard pruned db l1_batches up to {} and miniblocks up to {}, operation took {:?}",
+            current_pruning_info.last_soft_pruned_l1_batch.unwrap(),
+            current_pruning_info.last_soft_pruned_miniblock.unwrap(),
+            latency
+        );
+
+        Ok(())
+    }
+
+    pub async fn run_single_iteration(&self, pool: &ConnectionPool<Core>) -> anyhow::Result<bool> {
+        let mut storage = pool.connection().await?;
+        let current_pruning_info = storage.pruning_dal().get_pruning_info().await?;
+
+        if current_pruning_info.last_soft_pruned_l1_batch
+            == current_pruning_info.last_hard_pruned_l1_batch
+        {
+            let pruning_done = self.soft_prune(pool).await?;
+            if !pruning_done {
+                return Ok(false);
+            }
+
+            tokio::time::sleep(self.config.soft_and_hard_pruning_time_delta).await;
+        }
+
+        self.hard_prune(pool).await?;
+
+        Ok(true)
+    }
+    pub async fn run_in_loop(
+        self,
+        pool: ConnectionPool<Core>,
+        stop_receiver: watch::Receiver<bool>,
+    ) -> anyhow::Result<()> {
+        loop {
+            if *stop_receiver.borrow() {
+                tracing::info!("Stop signal received, shutting down DbPruner");
+            }
+            let _ = self.update_l1_batches_metric(&pool).await;
+            // as this component is not really mission-critical, all errors are generally ignored
```

Comment:
If we want to tolerate errors in the component, they at least should be logged with an appropriate level (and maybe converted to metrics, e.g. via an error counter).

Reply:
corrected